framework/src/org/apache/cordova/AllowListPlugin.java

```diff
@@ -0,0 +1,162 @@
+
+    // Used when instantiated via reflection by PluginManager
+    public AllowListPlugin() { }
+
+    // These can be used by embedders to allow Java-configuration of an allow list.
+    public AllowListPlugin(Context context) {
+        this(new AllowList(), new AllowList(), null);
+        new CustomConfigXmlParser().parse(context);
+    }
+
+    public AllowListPlugin(XmlPullParser xmlParser) {
+        this(new AllowList(), new AllowList(), null);
+        new CustomConfigXmlParser().parse(xmlParser);
+    }
+
+    public AllowListPlugin(AllowList allowedNavigations, AllowList allowedIntents, AllowList allowedRequests) {
+        if (allowedRequests == null) {
+            allowedRequests = new AllowList();
+            allowedRequests.addAllowListEntry("file:///*", false);
+            allowedRequests.addAllowListEntry("data:*", false);
+        }
+
+        this.allowedNavigations = allowedNavigations;
+        this.allowedIntents = allowedIntents;
+        this.allowedRequests = allowedRequests;
+    }
+
+    @Override
+    public void pluginInitialize() {
+        if (this.allowedNavigations == null) {
+            this.allowedNavigations = new AllowList();
+            this.allowedIntents = new AllowList();
+            this.allowedRequests = new AllowList();
+
+            new CustomConfigXmlParser().parse(webView.getContext());
+        }
+    }
+
+    private class CustomConfigXmlParser extends ConfigXmlParser {
+        @Override
+        public void handleStartTag(XmlPullParser xml) {
+            String strNode = xml.getName();
+            if (strNode.equals("content")) {
+                String startPage = xml.getAttributeValue(null, "src");
+                allowedNavigations.addAllowListEntry(startPage, false);
+            } else if (strNode.equals("allow-navigation")) {
+                String origin = xml.getAttributeValue(null, "href");
+                if ("*".equals(origin)) {
+                    allowedNavigations.addAllowListEntry("http://*/*", false);
+                    allowedNavigations.addAllowListEntry("https://*/*", false);
+                    allowedNavigations.addAllowListEntry("data:*", false);
+                } else {
+                    allowedNavigations.addAllowListEntry(origin, false);
+                }
+            } else if (strNode.equals("allow-intent")) {
+                String origin = xml.getAttributeValue(null, "href");
+                allowedIntents.addAllowListEntry(origin, false);
+            } else if (strNode.equals("access")) {
+                String origin = xml.getAttributeValue(null, "origin");
+                String subdomains = xml.getAttributeValue(null, "subdomains");
+                boolean external = (xml.getAttributeValue(null, "launch-external") != null);
+                if (origin != null) {
+                    if (external) {
+                        LOG.w(LOG_TAG, "Found <access launch-external> within config.xml. Please use <allow-intent> instead.");
```

Comment:
Now seems like a good time to drop this deprecated `<access launch-external>` support?

Reply:
Yes, we can remove this now too... I didn't carefully read to notice that this deprecated code existed. I will go ahead and fix this.

Reply:
@dpogue The launch-external has been removed: https://github.com/apache/cordova-android/pull/1138/commits/4fc48ff5093d0d8b5defc9fa9f58a72dc192d582?diff=split&w=1

Also moved `String subdomains` in the else statement where it is used.